nixos/modules/services/networking/pihole-ftl.nix

```diff
@@ -0,0 +1,102 @@
+{ pkgs
+, lib
+, config
+, ...
+}:
+
+with lib;
+
+let
+
+  cfg = config.services.pihole-ftl;
+  stateDir = "/etc/pihole";
+  confFile = "${stateDir}/pihole-FTL.conf";
+  logFile = "/var/log/pihole-FTL.log";
+
+in
+{
+
+  options.services.pihole-ftl = {
+    enable = mkEnableOption "Pi-hole FTL";
+
+    openFirewall = mkOption {
+      type = types.bool;
+      default = true;
+      description = "Open ports in the firewall for Pi-hole FTL.";
+    };
+
+    config = mkOption {
+      type = types.str;
+      default = "";
+      description = "Configuration file content for Pi-hole FTL.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pihole-ftl;
+      defaultText = "pkgs.pihole-ftl";
+      description = "Pi-hole FTL package to use.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.tmpfiles.rules = [
+      "d '${stateDir}' 0700 pihole pihole - -"
+      "f '${confFile}' 0400 pihole pihole - '${cfg.config}'"
+      "f '${logFile}' 0700 pihole pihole - -"
+    ];
+
+    # pihole-ftl conflicts with dnsmasq
+    services.dnsmasq.enable = false;
```

Comment:
Do we want to `lib.mkForce` this to false?

Reply:
Or assert so it's visible to the user?

Reply:
Ah, yes, that's a much, much better solution!

I completely forgot we had assertions haha. We definitely don't want to be *disabling* stuff in a user's config.